### DISCUSSION TITLE
How to fix: AttributeError: 'MySQLDialect_pymysql' object has no attribute 'case_sensitive'

### DISCUSSION THREAD

**Discussion by 0106WeiWeiDeng** [aio-libs/aiomysql#908] [Q&A] [answered]
Hi, all

I run the following code:
```
from aiomysql.sa import create_engine
import asyncio

async def go():
    engine = await create_engine(
        user="user",
        db="db",
        host="localhost",
        password="passwd",
    )
    async with engine.acquire() as conn:
        res = await conn.execute("SELECT * FROM user;")
        row = await res.fetchone()
        print(row["name"])
    engine.close()
    await engine.wait_closed()

asyncio.run(go())
```
but an error occurs, the error info is:

```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/deng/miniconda3/envs/shaoxia/lib/python3.9/asyncio/runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "/home/deng/miniconda3/envs/shaoxia/lib/python3.9/asyncio/base_events.py", line 647, in run_until_complete
    return future.result()
  File "<stdin>", line 9, in go
  File "/home/deng/miniconda3/envs/shaoxia/lib/python3.9/site-packages/aiomysql/sa/connection.py", line 179, in _execute
    ret = await create_result_proxy(
  File "/home/deng/miniconda3/envs/shaoxia/lib/python3.9/site-packages/aiomysql/sa/result.py", line 14, in create_result_proxy
    await result_proxy._prepare()
  File "/home/deng/miniconda3/envs/shaoxia/lib/python3.9/site-packages/aiomysql/sa/result.py", line 250, in _prepare
    self._metadata = ResultMetaData(self, cursor.description)
  File "/home/deng/miniconda3/envs/shaoxia/lib/python3.9/site-packages/aiomysql/sa/result.py", line 116, in __init__
    assert dialect.case_sensitive, \
AttributeError: 'MySQLDialect_pymysql' object has no attribute 'case_sensitive'
```

MySQL version: 8.0.29
System: Ubuntu 22.04 LTS
Python: 3.9.16
aiomysql: 0.1.1
PyMySQL: 1.0.2
SQLAlchemy: 2.0.5.post1

Does anyone has a clue to fix the error? Thanks in advance!

**Comment by 0106WeiWeiDeng** [comment]
Additional info: when I use SQLAlchemy to connect mysql and excute SQLs, no error occurs!

**Reply by 0106WeiWeiDeng** [reply]
Additional info:  use aiomysql to connect mysql and excute SQLs, no error occurs.

**Comment by 0106WeiWeiDeng** [comment, marked as answer]
just delete two line of code in aiomysql/sa/result.py, the code works fine! The code info:

https://github.com/aio-libs/aiomysql/blob/bb8697d827981fcbf24841e0990cccebadf49fbf/aiomysql/sa/result.py#L116
https://github.com/aio-libs/aiomysql/blob/bb8697d827981fcbf24841e0990cccebadf49fbf/aiomysql/sa/result.py#L117

**Comment by jurschreuder** [comment]
Is there a way to fix this without changing the library files? Else other developers cannot just `pip install` it anymore.

**Reply by 0106WeiWeiDeng** [reply]
It seems like that aiomysql.sa module has not been updated for a long time, I think maintainers should fix this error...

**Reply by Nothing4You** [reply]
SQLAlchemy 2.0 is not yet supported, unfortunately we don't even support 1.4 yet.
See #818

**Reply by 0106WeiWeiDeng** [reply]
oh, SQLAlchemy is a popular and convenient library, and aio-mysql has asyncio features. aiomysql.sa module is a very nice one which combined them together! I would appreciate it very much if this model can get updates.
